sdk/core/azure-core/azure/core/messaging.py

```diff
@@ -181,3 +181,16 @@ def from_dict(cls, event):
                     " The `source` and `type` params are required."
                     )
         return event_obj
+
+    @classmethod
+    def from_json(cls, json):
+        # type: (Any) -> CloudEvent
+        """
+        Returns the deserialized CloudEvent object when a json is provided.
+        :param json: The json string that should be converted into a CloudEvent. This can also be
+         a storage QueueMessage, eventhub's EventData or ServiceBusMessage
+        :type json: object
+        :rtype: CloudEvent
```

Comment:
Might be worth adding something like:
`:raises ValueError: If the provided JSON is invalid.`